11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+		m_scratchSemaphore = m_device->createSemaphore(0);
+
+		// Get compute queue
+		m_queue = getComputeQueue();
+		uint32_t queueFamilyIndex = m_queue->getFamilyIndex();
+
+		// Create a resettable command buffer
+		{
+			smart_refctd_ptr<nbl::video::IGPUCommandPool> cmdpool = m_device->createCommandPool(queueFamilyIndex, IGPUCommandPool::CREATE_FLAGS::RESET_COMMAND_BUFFER_BIT);
+			if (!cmdpool->createCommandBuffers(IGPUCommandPool::BUFFER_LEVEL::PRIMARY, 1u, &m_computeCmdBuf))
+				return logFail("Failed to create Command Buffers!\n");
+		}
+
+		// want to capture the image data upload as well
+		//m_api->startCapture();
+
+		// Load source and kernel images
+		{
+			IAssetLoader::SAssetLoadParams lp = {};
+			lp.logger = m_logger.get();
+			lp.workingDirectory = ""; // virtual root
+			auto srcImageBundle = m_assetMgr->getAsset("../../media/colorexr.exr", lp);
+			auto kerImageBundle = m_assetMgr->getAsset("../../media/kernels/physical_flare_256.exr", lp);
+			const auto srcImages = srcImageBundle.getContents();
+			const auto kerImages = kerImageBundle.getContents();
+			if (srcImages.empty() or kerImages.empty())
+				return logFail("Could not load image or kernel!");
+			auto srcImageCPU = IAsset::castDown<ICPUImage>(srcImages[0]);
+			auto kerImageCPU = IAsset::castDown<ICPUImage>(kerImages[0]);
+			const auto srcImageFormat = srcImageCPU->getCreationParameters().format;
+			const auto kerImageFormat = kerImageCPU->getCreationParameters().format;
+
+			// Create views for these images
+			ICPUImageView::SCreationParams viewParams[2] =
+			{
+				{
+					.flags = ICPUImageView::E_CREATE_FLAGS::ECF_NONE,
+					.image = std::move(srcImageCPU),
+					.viewType = IImageView<ICPUImage>::E_TYPE::ET_2D,
+					.format = srcImageFormat,
+					.subresourceRange = {
+						.aspectMask = IImage::E_ASPECT_FLAGS::EAF_COLOR_BIT,
+						.baseMipLevel = 0u,
+						.levelCount = ICPUImageView::remaining_mip_levels,
+						.baseArrayLayer = 0u,
+						.layerCount = ICPUImageView::remaining_array_layers
+					}
+				},
+				{
+					.flags = ICPUImageView::E_CREATE_FLAGS::ECF_NONE,
+					.image = std::move(kerImageCPU),
+					.viewType = IImageView<ICPUImage>::E_TYPE::ET_2D,
+					.format = kerImageFormat,
+					.subresourceRange = {
+						.aspectMask = IImage::E_ASPECT_FLAGS::EAF_COLOR_BIT,
+						.baseMipLevel = 0u,
+						.levelCount = ICPUImageView::remaining_mip_levels,
+						.baseArrayLayer = 0u,
+						.layerCount = ICPUImageView::remaining_array_layers
+					}
+			}
+			};
+			const auto srcImageViewCPU = ICPUImageView::create(std::move(viewParams[0]));
+			const auto kerImageViewCPU = ICPUImageView::create(std::move(viewParams[1]));
+
+			// Using asset converter
+			smart_refctd_ptr<nbl::video::CAssetConverter> converter = nbl::video::CAssetConverter::create({ .device = m_device.get(),.optimizer = {} });
+			// We don't want to generate mip-maps for these images (YET), to ensure that we must override the default callbacks.
```

Comment:
also hook the `ImageViewCPU`s into an `ICPUDescriptorSet` (create the CPU layout too), it will cut down your Descriptor Set creation code